src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/Completion/RemoteCompletionService.cs

```diff
@@ -0,0 +1,234 @@
+using Microsoft.CodeAnalysis.Razor.Protocol;
+using Microsoft.CodeAnalysis.Razor.Protocol.Completion;
+using Microsoft.CodeAnalysis.Razor.Remote;
+using Microsoft.CodeAnalysis.Remote.Razor.ProjectSystem;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Response = Microsoft.CodeAnalysis.Razor.Remote.RemoteResponse<Microsoft.VisualStudio.LanguageServer.Protocol.VSInternalCompletionList?>;
+using RoslynCompletionContext = Roslyn.LanguageServer.Protocol.CompletionContext;
+using RoslynCompletionSetting = Roslyn.LanguageServer.Protocol.CompletionSetting;
+
+namespace Microsoft.CodeAnalysis.Remote.Razor;
+
+internal sealed class RemoteCompletionService(in ServiceArgs args) : RazorDocumentServiceBase(in args), IRemoteCompletionService
+{
+    internal sealed class Factory : FactoryBase<IRemoteCompletionService>
+    {
+        protected override IRemoteCompletionService CreateService(in ServiceArgs args)
+            => new RemoteCompletionService(in args);
+    }
+
+    private readonly RazorCompletionListProvider _razorCompletionListProvider = args.ExportProvider.GetExportedValue<RazorCompletionListProvider>();
+    private readonly IClientCapabilitiesService _clientCapabilitiesService = args.ExportProvider.GetExportedValue<IClientCapabilitiesService>();
+
+    public ValueTask<CompletionPositionInfo?> GetPositionInfoAsync(
+        JsonSerializableRazorPinnedSolutionInfoWrapper solutionInfo,
+        JsonSerializableDocumentId documentId,
+        VSInternalCompletionContext completionContext,
+        Position position,
+        CancellationToken cancellationToken)
+        => RunServiceAsync(
+            solutionInfo,
+            documentId,
+            context => GetPositionInfoAsync(context, completionContext, position, cancellationToken),
+            cancellationToken);
+
+    private async ValueTask<CompletionPositionInfo?> GetPositionInfoAsync(
+        RemoteDocumentContext remoteDocumentContext,
+        VSInternalCompletionContext completionContext,
+        Position position,
+        CancellationToken cancellationToken)
+    {
+        var sourceText = await remoteDocumentContext.GetSourceTextAsync(cancellationToken).ConfigureAwait(false);
+        if (!sourceText.TryGetAbsoluteIndex(position, out var index))
+        {
+            return null;
+        }
+
+        var codeDocument = await remoteDocumentContext.GetCodeDocumentAsync(cancellationToken).ConfigureAwait(false);
+
+        var positionInfo = GetPositionInfo(codeDocument, index);
+
+        if (positionInfo.LanguageKind != RazorLanguageKind.Razor
+           && await DelegatedCompletionHelper.TryGetProvisionalCompletionInfoAsync(
+                remoteDocumentContext,
+                completionContext,
+                positionInfo,
+                DocumentMappingService,
+                cancellationToken)
+                .ConfigureAwait(false) is { } provisionalCompletionInfo)
+
```

Comment:
```suggestion
```

Reply:
Removed. FWIW, Format Document doesn't fix this.